bellboy/sensors.py

```diff
@@ -0,0 +1,145 @@
+# Sensor classes
+# Sensors include Camera, Ultrasonicm Mic....
+import logging
+import time
+from threading import Thread
+
+import picamera
+import RPi.GPIO as GPIO
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractSensor():
+
+    # constructor.
+    # creates private variables for:
+    #   the sensor's data buffer,
+    #   the polling period, and
+    #   the sensor's polling thread
+    # @param poll_period_us time inbetween sensor reads, in microseconds
+    def __init__(self, poll_period_us):
+
+        self._poll_period = poll_period_us
+        self._buffer = []
+        super().__init__()
+        self._sensor_thread = Thread(target=self._read_sensor_loop)
+
+    # setup the sensor
+    def init_sensor(self):
+        pass
+
+    # close the sensor
+    def close_sensor(self):
+        pass
+
+    # string rep of sensors current status
+    def brief_status(self):
+        pass
+
+    # loop to be run in sensor's thread
+    def _read_sensor_loop(self):
+        # wait for self._poll_period
+        pass
+
+    def join(self):
+        self._sensor_thread.join()
+
+    def begin(self):
+        logger.debug("starting ultrasonic sensor thread")
+        self._sensor_thread.start()
+
+
+class TestSensor(AbstractSensor):
+    def __init__(self):
```

Comment:
Codacy found an issue: [Missing docstring in __init__ (D107)](https://app.codacy.com/gh/Bellboy-Capstone/System/pullRequest?prid=6272175)